test/ngAnimate/ngSwapSpec.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+describe("ngSwap", function() {
+
+  beforeEach(module('ngAnimate'));
+  beforeEach(module('ngAnimateMock'));
+
+  var element;
+  afterEach(inject(function($$jqLite) {
+    dealoc(element);
+  }));
+
+  it('should render a new container when the expression changes', inject(function($rootScope, $compile, $animate) {
+    $animate.enabled(false);
+
+    element = $compile('<div><div ng-swap="exp">{{ exp }}</div></div>')($rootScope);
+    $rootScope.$digest();
+
+    var first = element.find('div')[0];
+    expect(first).toBeFalsy();
+
+    $rootScope.exp = 'yes';
+    $rootScope.$digest();
+
+    var second = element.find('div')[0];
+    expect(second.innerText).toBe('yes');
+
+    $rootScope.exp = 'super';
+    $rootScope.$digest();
+
+    var third = element.find('div')[0];
+    expect(third.innerText).toBe('super');
+    expect(third).not.toEqual(second);
+    expect(second.parentNode).toBeFalsy();
+  }));
+
+  it('should render a new container only when the expression property changes',
+    inject(function($rootScope, $compile, $animate) {
+
+    $animate.enabled(false);
+
+    element = $compile('<div><div ng-swap="exp.prop">{{ exp.value }}</div></div>')($rootScope);
+    $rootScope.exp = {
+      prop: 'hello',
+      value: 'world'
+    };
+    $rootScope.$digest();
+
+    var one = element.find('div')[0];
+    expect(one.innerText).toBe('world');
+
+    $rootScope.exp.value = 'planet';
+    $rootScope.$digest();
+
+    var two = element.find('div')[0];
+    expect(two.innerText).toBe('planet');
+    expect(two).toBe(one);
+
+    $rootScope.exp.prop = 'goodbye';
+    $rootScope.$digest();
+
+    var three = element.find('div')[0];
+    expect(three.innerText).toBe('planet');
+    expect(three).not.toBe(two);
+  }));
+
+  it('should watch the expression as a collection',
+    inject(function($rootScope, $compile, $animate) {
+
+    $animate.enabled(false);
+
+    element = $compile('<div><div ng-swap="exp">{{ exp.a }} {{ exp.b }} {{ exp.c }}</div></div>')($rootScope);
+    $rootScope.exp = {
+      a: 1,
+      b: 2
+    };
+    $rootScope.$digest();
+
+    var one = element.find('div')[0];
+    expect(one.innerText.trim()).toBe('1 2');
+
+    $rootScope.exp.a++;
+    $rootScope.$digest();
+
+    var two = element.find('div')[0];
+    expect(two.innerText.trim()).toBe('2 2');
+
+    $rootScope.exp.c = 3;
+    $rootScope.$digest();
+
+    var three = element.find('div')[0];
+    expect(three.innerText.trim()).toBe('2 2 3');
+
+    $rootScope.exp = { c: 4 };
+    $rootScope.$digest();
+
+    var four = element.find('div')[0];
+    expect(four.innerText.trim()).toBe('4');
+  }));
```

Comment:
I don't think this test tests what it claims it tests.
Basically, what you want to test is that elements are swapped, not just that the `textContent` updates based on `exp`.
You should test that `one` !== `two` !== `three`.